frontend/src/main/java/com/google/tripmeout/frontend/storage/InMemoryPlaceVisitStorage.java

```diff
@@ -1,24 +1,24 @@
 package com.google.tripmeout.frontend.storage;
 
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import com.google.tripmeout.frontend.error.PlaceVisitAlreadyExistsException;
+import com.google.tripmeout.frontend.error.PlaceVisitNotFoundException;
+import com.google.tripmeout.frontend.error.TripNotFoundException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
-
-import com.google.tripmeout.frontend.PlaceVisitModel;
-import com.google.tripmeout.frontend.error.PlaceVisitAlreadyExistsException;
-import com.google.tripmeout.frontend.error.PlaceVisitNotFoundException;
-import com.google.tripmeout.frontend.error.TripNotFoundException;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 public class InMemoryPlaceVisitStorage implements PlaceVisitStorage {
   // <tripId, placeId, PlaceVisitModel>
-  Map<String, Map<String, PlaceVisitModel>> storage = new ConcurrentHashMap<>();
+  Map<String, Map<String, PlaceVisitModel>> placesByPlaceIdByTripId = new ConcurrentHashMap<>();
 
   @Override
   public void addPlaceVisit(PlaceVisitModel placeVisit) throws PlaceVisitAlreadyExistsException {
     try {
-      storage.compute(placeVisit.tripId(), (tripKey, placesMap) -> {
+      placesByPlaceIdByTripId.compute(placeVisit.tripId(), (tripKey, placesMap) -> {
```

Comment:
Nit: “by trip ID, by place ID” makes more sense since this is the order of access (it currently reads in an inverted fashion).